backend/repository/endpoint_update.go

```diff
@@ -82,6 +83,25 @@ func (r *Repository) validateEndpointCfg(req *EndpointConfig) error {
 	} else {
 		return errors.Errorf("client type %q is not supported", clientCfg.Type)
 	}
+
+	for _, mid := range req.Middlewares {
+		for _, midCfg := range r.gatewayConfig.Middlewares {
+			if mid.Name == midCfg.Name {
+				result, err := r.validator.ValidateGo(midCfg.SchemaFile, mid.Options)
```

Comment:
The error handling logic here is complex. It is better to hide these `if` inside `ValidteGo` and define it as

`func ValidateSchema(schemaFile string, arbitraryObj map[string]interface{}) error`

Reply:
ok.  i was using the gojsonschema API as a guideline.  It seems they chose to distinguish general errors (file read, JSON parsing  etc)  from validation errors which are properties on the Result object.  But I am ambivalent and your suggestion would keep this code simpler.

Reply:
ok. i was using the gojsonschema API as a guideline. It seems they chose to distinguish general errors (file read, JSON parsing etc) from validation errors which are properties on the Result object. But I am ambivalent and your suggestion would keep this code simpler.